From 9f984c0235c92c062bed66ca84c7b246d0815168 Mon Sep 17 00:00:00 2001
From: Brad Embree <brad@bestpractical.com>
Date: Thu, 12 Jun 2025 15:54:54 -0700
Subject: [PATCH] Split multiple tomselect initial value on delimiter

When setting the initial values for a multiple value tomselect control the
values need to be split on the delimiter to avoid adding a value that includes
all delimited values.

Since ", " is the delimiter for multiple value inputs, we should process
default values in EditCustomFieldAutocomplete in the same way. This fixes
autocomplete's default values when creating a related ticket from Linked
widget.
---
 share/html/Elements/EditCustomFieldAutocomplete |  2 +-
 share/static/js/util.js                         | 16 ++++++++++++++--
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/share/html/Elements/EditCustomFieldAutocomplete b/share/html/Elements/EditCustomFieldAutocomplete
index 154ae2c218..6320b768ea 100644
--- a/share/html/Elements/EditCustomFieldAutocomplete
+++ b/share/html/Elements/EditCustomFieldAutocomplete
@@ -55,7 +55,7 @@ data-options-source="<% $session{CurrentUser}->Privileged ? '' : '/SelfService'
 value="<% $Default || '' %>" />
 <%INIT>
 my $name = $Name || $NamePrefix . $CustomField->Id . ( $Multiple ?  '-Values' : '-Value' );
-if ( $Default && !$Multiple ) {
+if ( $Default ) {
     $Default =~ s/\s*\r*\n\s*/,  /g; # 2 spaces here are on purpose, in case the value contains ", "
 }
 if ( $Multiple and $Values ) {
diff --git a/share/static/js/util.js b/share/static/js/util.js
index fe22e0e122..5a5cef8826 100644
--- a/share/static/js/util.js
+++ b/share/static/js/util.js
@@ -452,12 +452,24 @@ function initializeSelectElement(elt) {
         }
     }
 
-    const value = elt.value || elt.getAttribute('data-value');
     new TomSelect(elt,settings);
 
     // If the default value is not in the options, add it.
+    const value = elt.value || elt.getAttribute('data-value');
     if ( value ) {
-        (Array.isArray(value) ? value : [value]).forEach(value => {
+        let values = [];
+        if ( Array.isArray(value) ) {
+            values = value;
+        }
+        else {
+            if ( elt.hasAttribute('data-autocomplete-multiple') ) {
+                values = value.split(",  ");
+            }
+            else {
+                values = [ value ];
+            }
+        }
+        values.forEach(value => {
             if ( !elt.tomselect.getItem(value) ) {
                 elt.tomselect.createItem(value, true);
                 elt.tomselect.addItem(value, true);
-- 
2.25.1

